### DISCUSSION TITLE
Just found I can't retrieve Custom Metadata Records

### DISCUSSION THREAD

**Discussion by waynechungvs** [jetstreamapp/jetstream#684] [General]
I've tested this amazing tool for a while but just found a bug ( or I missed something, please everyone to let me know), I can retrieve Custom Metadata object but just can't retrieve Custom Metadata records.

**Comment by paustint** [comment]
@waynechungvs - For custom metadata, you have two options for getting records:

1. Use Query Builder to query custom metadata records, and from there you can use the data loader to upsert custom metadata records
2. Use the Deploy and compare metadata to work with the records

Let me know if this solves your question.

### Configuration

![image](https://github.com/jetstreamapp/jetstream/assets/5461649/52271436-5930-407e-9280-3f23146d3e91)

![image](https://github.com/jetstreamapp/jetstream/assets/5461649/304124fa-96a7-4e72-9300-cccd03ffe067)

### Compare / Deploy Metadata

![image](https://github.com/jetstreamapp/jetstream/assets/5461649/868ffa35-b600-4b69-b746-bb44aa025f7a)

### Query Custom Metadata

![image](https://github.com/jetstreamapp/jetstream/assets/5461649/394b2392-0b4a-48a8-9cc3-119fe234a0a3)

![image](https://github.com/jetstreamapp/jetstream/assets/5461649/76cafc3a-220b-48f4-8180-6ac2bc2dc6f5)

**Reply by waynechungvs** [reply]
Hi @paustint Thanks for rapid response. I attached few screenshot to clarify if I missed some steps please let me know.
And my Jetstream is connect to correct org which has custom metadata records show in image 2 and 3. 
![image](https://github.com/jetstreamapp/jetstream/assets/105181305/1fc85996-d2d8-4258-9b4c-9910990919e8)
![image](https://github.com/jetstreamapp/jetstream/assets/105181305/2a4ef490-c967-4168-85ae-6754a6091f42)
![image](https://github.com/jetstreamapp/jetstream/assets/105181305/22f56727-24da-4653-8b6a-01ccf483c935)

**Reply by paustint** [reply]
@waynechungvs - I see, It looks like for some reason Salesforce does not properly keep track of the modified date for custom metadata when using their metadata api, which is what powers this feature, so apparently the data filter for custom metadata will not be reliable.

![image](https://github.com/jetstreamapp/jetstream/assets/5461649/ffd3f5d1-3df4-4436-b499-637cf078f458)


Oddly enough, if you query the custom records there is a seemingly accurate timestamp - so your workaround here be to use the query tool to see what was modified with the dates so you can see what data you may need to work with.

It might be possible for Jetstream to do some extra work to query the records and then apply those updates to the metadata api response so that this feature is able to work as expected.

The Salesforce Metadata API has a lot of quirks, as I am sure you have noticed if you have spent any amount of time working with change sets.

![image](https://github.com/jetstreamapp/jetstream/assets/5461649/c3f89740-6611-4ad5-a350-401cf0fa77d7)
